library/channel.c

```diff
@@ -250,6 +250,11 @@ bool ziti_channel_is_connected(ziti_channel_t *ch) {
     return ch->state == Connected;
 }
 
+static void ziti_channel_set_api_session_id(ziti_channel_t *ch, const char* api_session_id) {
+    memset(ch->api_session_id, '\0', sizeof(ch->api_session_id));
+    strncpy(ch->api_session_id, api_session_id, strlen(api_session_id));
```

Comment:
should be `strncpy(ch->api_session_id, api_session_id, sizeof(ch->api_session_id))` in case passed in value is longer than expected

Reply:
changed